### PR TITLE
`azurerm_kubernetes_cluster` `azurerm_kubernetes_cluster_node_pool` -  `host_group_id`

### PR DESCRIPTION
**Pull request by shu-ying789**
linked to https://github.com/hashicorp/terraform-provider-azurerm/issues/15658

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @shu-ying789, this looks good to me. Two minor comments on the docs and there is also a test failure, looks like the location is passed to `kubernetes_version`?
```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_dedicatedHost
=== PAUSE TestAccKubernetesCluster_dedicatedHost
=== CONT  TestAccKubernetesCluster_dedicatedHost
testcase.go:110: Step 1/1 error: Error running apply: exit status 1
Error: creating Cluster: (Managed Cluster Name "acctestaks220428103126887507" / Resource Group "acctestRG-aks-220428103126887507"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Client Error: error parsing version(westus2) as semver: Invalid Semantic Version"
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 39, in resource "azurerm_kubernetes_cluster" "test":
39: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccKubernetesCluster_dedicatedHost (320.81s)
FAIL
```

File: website/docs/d/kubernetes_cluster.html.markdown [hunk 1]
```diff
@@ -122,6 +122,8 @@ A `agent_pool_profile` block exports the following:
 
 * `enable_node_public_ip` - If the Public IPs for the nodes in this Agent Pool are enabled.
 
+* `host_group_id` - The ID of a Dedicated Host Group that this xxx should be run within. Changing this forces a new resource to be created.
```
[the remark this answers is not in the document]
Comment (shu-ying789) [reply website/docs/d/kubernetes_cluster.html.markdown]: @stephybun Documentation and test have been modified, please check again, thank you!

**Review by stephybun — COMMENTED** [review]
Have another test failure:
```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_dedicatedHost
=== PAUSE TestAccKubernetesCluster_dedicatedHost
=== CONT  TestAccKubernetesCluster_dedicatedHost
testcase.go:110: Step 1/1 error: Error running apply: exit status 1
Error: waiting for creation of Cluster: (Managed Cluster Name "acctestaks220512100349784765" / Resource Group "acctestRG-aks-220512100349784765"): Code="CreateVMSSAgentPoolFailed" Message="Code=\"AllocationFailed\" Message=\"VM allocation to the dedicated host failed. Please ensure that the dedicated host has enough capacity or try allocating elsewhere.\""
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 39, in resource "azurerm_kubernetes_cluster" "test":
39: resource "azurerm_kubernetes_cluster" "test" {
testing_new.go:70: Error running post-test destroy, there may be dangling resources: exit status 1
Error: deleting Dedicated Host Group "Host Group: (Name \"acctestDHG-compute-220512100349784765\" / Resource Group \"acctestRG-aks-220512100349784765\")" : compute.DedicatedHostGroupsClient#Delete: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="OperationNotAllowed" Message="Host Group 'acctestDHG-compute-220512100349784765' cannot be deleted. Before deleting the host group, please ensure it does not contain any Host. Also, the host group should not contain any Virtual Machine or Virtual Machine Scale Set placed through automatic placement." Target="hostGroup"
--- FAIL: TestAccKubernetesCluster_dedicatedHost (401.57s)
FAIL
```

**Comment by ms-henglu** [comment]
Hi @stephybun - thanks for code review, there seem to be an issue that the resources can't be destroyed successfully, I'll reopen this PR when it's fixed.

**Pull request closed without merging**